Dockerfile

```
@@ -1,8 +1,9 @@
-FROM python:3.5
+FROM python:3.8.6
 ENV WEBHOOK_URL_PREFIX="wiki/hooks"
+ADD requirements.txt .
 RUN apt update
 RUN apt install -y unzip apache2 build-essential python3-dev python3-pip python3-setuptools python3-wheel python3-cffi libcairo2 libpango-1.0-0 libpangocairo-1.0-0 libgdk-pixbuf2.0-0 libffi-dev shared-mime-info
-RUN python -m pip  install mkdocs==1.0.4 mkdocs-material==3.1.0 pygments pymdown-extensions mkdocs-pdf-export-plugin==0.4.0
+RUN python -m pip  install -r requirements.txt
```

If you add the requirements at the beginning it will invalidate the django cache everytime the requirements changes and so docker needs to reinstall the 
Run apt update and the
Run apt install part

```suggestion
ADD requirements.txt .
RUN python -m pip  install -r requirements.txt
```